### DISCUSSION TITLE
[Question] Undo manager and `setStyle`

### DISCUSSION THREAD

**Discussion by iamkeiryan** [GrapesJS/grapesjs#6299] [Q&A]
hi @artf, 

I have created a custom style manager that uses `setStyle` to update css in the canvas. Whenever I try to undo it after a single change, the css are getting removed completely. I haven't dig deeper yet with the undo manager and was hoping that this is a configuration issue.

Here's the edited canvas:
![image](https://github.com/user-attachments/assets/4c75ea1d-9257-4eeb-ad3c-2183d00c4053)
Here's what happens after undo: 
![image](https://github.com/user-attachments/assets/03907fa8-82d3-46ad-a3a2-cc13efac0c6e)

I'm using the `editor?.runCommand('core:undo');` command. 

Edit:
Upon checking the details, the css is getting removed completely whenever I checked the `editor.getCSS`. Haven't found a solution on what causes this.

**Comment by artf** [comment]
@iamkeiryan a reproducible demo would help here
